pkg/platform/runtime/setup/setup.go

```diff
@@ -382,19 +382,40 @@ func (s *Setup) fetchAndInstallArtifactsFromBuildPlan(installFunc artifactInstal
 
 	// Compute and handle the change summary
 	// runtimeAndBuildtimeArtifacts records all artifacts that will need to be built in order to obtain the runtime.
-	// Disabled due to DX-2033.
-	// Please use this var when we come back to this in the future as we need to make a clear distinction between this
-	// and runtime-only artifacts.
-	// var runtimeAndBuildtimeArtifacts artifact.Map
+	var runtimeAndBuildtimeArtifacts artifact.Map
 	var runtimeArtifacts artifact.Map // Artifacts required for the runtime to function
-	if buildResult.Build != nil {
+
+	// If the build is not ready, we need to get the runtime and buildtime closure
+	if !buildResult.BuildReady {
+		runtimeAndBuildtimeArtifacts, err = buildplan.NewBuildtimeMapFromBuildPlan(buildResult.Build)
+		if err != nil {
+			return nil, nil, errs.Wrap(err, "Failed to create artifact map from build plan")
+		}
+	}
+
+	// If we are installing build dependencies, then buildtime dependences are also runtime dependencies
+	if strings.EqualFold(os.Getenv(constants.InstallBuildDependencies), "true") {
+		logging.Debug("Installing build dependencies")
+		if runtimeAndBuildtimeArtifacts == nil {
+			runtimeAndBuildtimeArtifacts, err = buildplan.NewBuildtimeMapFromBuildPlan(buildResult.Build)
+			if err != nil {
+				return nil, nil, errs.Wrap(err, "Failed to create artifact map from build plan")
+			}
+		}
+		runtimeArtifacts = runtimeAndBuildtimeArtifacts
+	} else {
 		runtimeArtifacts, err = buildplan.NewMapFromBuildPlan(buildResult.Build)
 		if err != nil {
 			return nil, nil, errs.Wrap(err, "Failed to create artifact map from build plan")
 		}
 	}
 
-	setup, err := s.selectSetupImplementation(buildResult.BuildEngine, runtimeArtifacts)
+	var setup Setuper
+	if !buildResult.BuildReady {
+		setup, err = s.selectSetupImplementation(buildResult.BuildEngine, runtimeAndBuildtimeArtifacts)
+	} else {
+		setup, err = s.selectSetupImplementation(buildResult.BuildEngine, runtimeArtifacts)
+	}
```

Comment:
It does not, however, because the setup implementation is responsible for resolving artifact names if we do not pass it the correct artifact map we will get a bunch of `unknown` artifact progress bars.

Reply:
Ok, that seems like a problem worth addressing, but let's address it explicitly rather than implicitly. This is breaking the contract I worked hard to set up because of all the issues we ran into by being so implicit.

Reply:
Fair enough. I've move the name resolution out to it's own type that is responsible for just that.